doc/developer/design/20230223_stabilize_with_mutually_recursive.md

```diff
@@ -0,0 +1,261 @@
+Test scenarios can be categorized along two dimensions:
+
+### By type
+
+1. **Unit tests.**
+We aim to have one unit test per transform.
+We can invest time proportional to the complexity of the transform to ensure that each transform is correct.
+
+1. **Integration tests.**
+We will add test scenarios inspired by the use cases of our prospects as end-to-end `*.slt` tests.
+We will also add at least one long-running `mzcompose` test runs [as part of our nightly tests](https://buildkite.com/materialize/nightlies) and is used when qualifying future releases.
+
+As those tests will include expected results, it will be great if we have a reference external iteration driver for the semantics proposed in the original design doc[^wmr].
+That way we can cross-check the results of the reference and the internal implementation of WMR support and ensure that both produce equal results.
+We can implement such driver in Python and integrate it in our `mzcompose` tests.
+
+### By test scenario
+
+1. **Synthetic tests.**
+The best synthetic use case that we have identified so far seems to be the LDBC social network benchmark[^ldbc].
+Although there is a dedicated epic for that (MaterializeInc/materialize#17591), we can a selected subset of the work in order to bootstrap a testing environment that consists of 
+  (a) LDBC data + updates, and 
+  (b) several of the recursive queries defined by the benchmark. 
+We can use the choke-point characterization of each query to figure out the most representative subset.
+
+1. **Use-case driven.**
+It is unclear how useful these will be as load tests, as we don't have the resources to write realistic data generators to replicate the domain of specific customers. 
+However, we can try to map some of the customer use cases to the LDBC dataset.
+
+1. **Sourced from elsewhere.**
+We can check what tests Postgres has for their `WITH RECURSIVE` support.
+
+## Lifecycle
+[lifecycle]: #lifecycle
+<!--
+If the design is risky or has the potential to be destabilizing, you should plan to roll the implementation out behind a feature flag.
+List all feature flags, their behavior and when it is safe to change their value.
+Describe the [lifecycle of the feature](https://www.notion.so/Feature-lifecycle-2fb13301803b4b7e9ba0868238bd4cfb).
+Will it start as an alpha feature behind a feature flag?
+What level of testing will be required to promote to beta?
+To stable?
+-->
+
+We plan to roll the implementation behind a `with_mutually_recursive` feature flag.
+It should be OK to turn the feature flag on for individual environments at all times.
+It should be OK to turn the feature flag off for customers as long as they don't have catalog objects that use the feature.
+
+The feature will go through an `alpha`/`beta`/`stable` lifecycle. 
+The conditions for `alpha` stage are assumed to be already implicitly satisfied because `staging ` environments have access to the feature (as they operate under `--unsafe-mode`).
```

Comment:
I don't think that is true? You still need to enable it manually: https://www.notion.so/materialize/Debugging-Materialize-Cloud-0788c1fe692b4acf9d247345e7b5346e?pvs=4#b2da86e74f814f94bdfcbf8cd423720a